numpy/core/src/multiarray/refcount.c

```diff
@@ -204,6 +296,10 @@ PyArray_INCREF(PyArrayObject *mp)
 /*NUMPY_API
   Decrement all internal references for object arrays.
   (or arrays with object fields)
+
+  The use of this function is strongly discouraged, within NumPy
+  use PyArray_Clear, which DECREF's and sets everything to NULL and can
+  work with any dtype.
```

Comment:
Should (can?) `PyArray_XDECREF` be deprecated?

Reply:
Probably, not sure there is a need for it to begin in reasonable code... But seems hard to do, might be a "nuke it with NumPy 2.0" thing...